mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java

```diff
@@ -119,7 +120,9 @@
         return Flux.from(messageMetadataVault.listMessages(bucketName, user))
             .filter(deletingInformation -> messageId.equals(deletingInformation.getDeletedMessage().getMessageId()))
             .flatMap(deletingInformation -> deleteMessageData(bucketName, deletingInformation)
-                .then(deleteMessageMetaData(bucketName, user, messageId)))
+                .then(deleteMessageMetaData(bucketName, user, messageId))
+                .thenReturn(deletingInformation))
+            .flatMap(this::deleteSameBlobInOtherBuckets)
```

Comment:
If I understand well, you : 
* list all buckets
* list the messages of the user in each bucket
* filter out the messages not equal to the MessageId
* then for each of those messages : 
  * delete it in the current searched bucket
  * delete its metadata
* then delete that same blob potentially present in other buckets.

Why not just deleting the same blob in all buckets directly in one place, before deleting the metadata, instead of dividing that in two operations? Or is there something better? I feel it could be easier

Reply:
Regards to the complex of code to write, I think they are the same, about the performance, two ways taking the same cost:
 - need to delete the metadata message
 - need to delete blobs of the message(its own blob and other blobs somewhere else)

I prefer the following story: I delete my self data, then I delete other people data
 - delete the message (blob + metadata)
 - delete the related blobs.

Rather than mixing:
 - delete all blobs (included the message blob)
 - delete the message metadata.

Reply:
Other people idea?